homeassistant/components/spotify/browse_media.py

```diff
@@ -163,6 +196,16 @@ async def async_browse_media(
         can_play_artist=can_play_artist,
     )
 
+    # Build new URLs with config entry specifyers
+    if parsed_url is not None:
```

Comment:
Why only do that with parsed_url? We should just always prefix the URLs or else you add a 2nd Spotify account and your URLs might now be resolved to the other config entry.

Reply:
The config entry is added to the root nodes regardless of if there's one or multiple Spotify accounts.

The idea was to be backwards compatibility with custom components using the 2022.2 API which don't pass `None, None` to `async_browse_media` to get the root nodes, and don't know or about the URLs and just expect plain Spotify URIs as `media_content_id`.

Maybe that's not a concern though, so we should only accept `None`, `None` or a `media_content_id` which is a `spotify://` URL with config entry + spotify URI, and fail otherwise?

Reply:
Yeah, we can just fail. I doubt custom component authors jumped on this.